packages/repo/tests/sync/checkout.test.ts

```diff
@@ -0,0 +1,56 @@
+import * as crypto from '@atproto/crypto'
+import { DidResolver } from '@atproto/did-resolver'
+import { Repo, RepoContents } from '../../src'
+import { MemoryBlockstore } from '../../src/storage'
+import * as sync from '../../src/sync'
+
+import * as util from '../_util'
+
+describe('Checkout Sync', () => {
+  let storage: MemoryBlockstore
+  let syncStorage: MemoryBlockstore
+  let repo: Repo
+  let keypair: crypto.Keypair
+  let repoData: RepoContents
+  const didResolver = new DidResolver()
+
+  beforeAll(async () => {
+    storage = new MemoryBlockstore()
+    keypair = await crypto.Secp256k1Keypair.create()
+    repo = await Repo.create(storage, keypair.did(), keypair)
+    syncStorage = new MemoryBlockstore()
+    const filled = await util.fillRepo(repo, keypair, 20)
+    repo = filled.repo
+    repoData = filled.data
+  })
+
+  it('sync a non-historical repo checkout', async () => {
+    const checkoutCar = await sync.getCheckout(storage, repo.cid)
+    const checkout = await sync.loadCheckout(
+      syncStorage,
+      checkoutCar,
+      didResolver,
+    )
+    const checkoutRepo = await Repo.load(syncStorage, checkout.root)
+    const contents = await checkoutRepo.getContents()
+    expect(contents).toEqual(repoData)
+    expect(checkout.contents).toEqual(repoData)
+  })
+
+  it('does not sync unneeded blocks during checkout', async () => {
+    const commitPath = await storage.getCommitPath(repo.cid, null)
+    if (!commitPath) {
+      throw new Error('Could not get commitPath')
+    }
+    const hasGenesisCommit = await syncStorage.has(commitPath[0])
+    expect(hasGenesisCommit).toBeFalsy()
+  })
+
+  it('throws on a bad signature', async () => {
+    const badRepo = await util.addBadCommit(repo, keypair)
+    const checkoutCar = await sync.getCheckout(storage, badRepo.cid)
+    await expect(
+      sync.loadCheckout(syncStorage, checkoutCar, didResolver),
+    ).rejects.toThrow()
```

Comment:
Might be worth verifying what is thrown here.  I think this would do the trick:
```suggestion
    ).rejects.toThrow('Invalid signature on commit')
```

Reply:
Ah, just noticed `RepoVerificationError` which is probably what we'd actually want here.

Reply:
ah good looks 👍